config/routes.yml

```diff
@@ -6,36 +6,120 @@ home:
   params: []
 
 blog:
-  uri: ^/blog$
+  uri: ^/blog/?$
```

Comment:
Inverse order of route. Always take the more precise first, here blog_single_post and after blog